pkg/providers/imagefamily/nodeimage.go

```diff
@@ -38,10 +38,8 @@ type NodeImageProvider interface {
 
 // Returns the list of available NodeImages for the given AKSNodeClass sorted in priority ordering
 func (p *Provider) List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) ([]NodeImage, error) {
-	kubernetesVersion, err := p.KubeServerVersion(ctx)
-	if err != nil {
-		return []NodeImage{}, err
-	}
+	// Should we be nil checking status?
```

Comment:
Don't need to, as both Status, and KubernetesVersion are values not pointers. However, we should check the Status Condition for KubernetesVersion, and/or the value != "".